bin/cisc_to_adie_conversion.py

```
@@ -0,0 +1,169 @@
+"""
+#!/usr/bin/env/ python 
+
+Author: Will Strawson and Hao-Ting Wang, last updated 22-02-2021
+Aim of this script is to convert CISC ID to ADIE ID in directory and file names 
+
+
+Usage:
+When running from the command line, the user is asked to input the parent directory which
+contains the subject directories they wish to convert.
+"""
+
+import pandas as pd
+import os 
+import re
+import shutil
+import sys, os
+from pathlib import Path
+import glob
+
+
+# Script has to run from within critchely_adie/ cisc2 dir for the paths to work
+# TODO: construct CISC2 file path
+# TODO: change session- labels to be 'baseline' or 'intervention'
+
+# Get path to ADIE dir - this is universal and should work for anyone running on the 
+# SN (Sussex neuroscience) server
+adie_dir = ('/research/cisc2/projects/critchley_adie/')
+
+# import conversion txt file 
+txtfile = os.path.join(adie_dir, 'BIDS_data/sourcedata/adie_idconvert.txt')
+print ('Conversion txt file path =',txtfile)
+
+# convert this to dictonary, where key = CISC and val = ADIE 
+def convert_dict(txtfile):  
+    with open(txtfile) as f:
+        rename = {}
+        for line in f:
+            (key,val)=line.split()
+            #Remove any non-digit i.e. 'CISC'
+            rename[str(re.sub("[^0-9]","",key))] = str(val)
+        return rename
+
+# Function 1 - consolidate sub- dir as string - bit redundent for now but may come in handy 
+def subpaths(sub):
+    return str(sub)
+
+# Function 2 - Extract CISC ID from directory, and match with ADIE ID 
+def idmatch(r):
+    # search for group of numbers after 'sub-'
+    print(r)
+    cid = re.search("sub-([0-9]*)",r).group(1)
+    # match CISC ID with ADIE ID
+    if cid in rename.keys():
+        # return new ID with 'sub-' appended, as per BIDS convention, and CISCID
+        return "sub-" + rename[cid], cid
+    elif cid not in rename.keys():
+        print("CISC ID ({}) not recognized!".format(cid))
```

I would use `raise` for errors instead of print.

---

Cool, will start using this